packages/amplify-e2e-core/src/utils/sdk-calls.ts

```diff
@@ -204,6 +204,38 @@ export const describeCloudFormationStack = async (stackName: string, region: str
   );
 };
 
+export const getNestedStackID = async (stackName: string, region: string, logicalId: string) => {
+  const cfnClient = new CloudFormation({ region });
+  return (await cfnClient.describeStackResources({ StackName: stackName, LogicalResourceId: logicalId }).promise()).StackResources[0]
```

Comment:
Potentially you could get an undefined reference here

Reply:
Agree, enhance the check for the result of the API call.

Reply:
I think this is fine if It's undefined then the test should fail anyway.

Reply:
No it is not, none of us would like to troubleshoot `undefined` errors I guess.